tests/codegen/params_count.rs

```diff
@@ -0,0 +1,24 @@
+//! Test that the generated code has the right number of parameters
+
+#![feature(no_core)]
+#![no_core]
+#![no_main]
+
+extern crate mini_core;
+
+// CHECK-LABEL: foo
+// CHECK-LABEL: main
+
+// expect three int params
+// CHECK-LABEL: foo
+// CHECK: (int32_t {{[a-zA-Z_][a-zA-Z0-9_]*}}, int32_t {{[a-zA-Z_][a-zA-Z0-9_]*}}, int32_t {{[a-zA-Z_][a-zA-Z0-9_]*}})
```

Comment:
Does this work? I don't think there is any need to assert the actual pattern of the variable names

```suggestion
// CHECK: (int32_t {{ \w* }}, int32_t {{ \w* }}, int32_t {{ \w* }})
```

Reply:
Neither `\w*` nor `[\w\d_]*` works for me.

Reply:
Ah yeah, filecheck uses [ERE](https://en.wikibooks.org/wiki/Regular_Expressions/POSIX-Extended_Regular_Expressions). I think something like `[[:alnum:]_]*` should work but if not, just leave it as `[a-zA-Z0-9_]*` (I think it's okay to not check that the first character isn't a number)